### PR TITLE
Onboarding: Login flow

### PR DESCRIPTION
**Pull request by vvaditya23**
**This PR contains welcome back screen, OTP screen and Password screen.**
_Note: Comments from [previous PR](https://github.com/internlabs/iOSProjectShaadi/pull/1) are addressed here!_

https://github.com/user-attachments/assets/42dd8d13-7af4-40bb-9faf-41a9f89808a2

### CODE REVIEW

File: iOSProjectShaadi.xcodeproj/project.xcworkspace/contents.xcworkspacedata
Comment (Pulkit29) [inline iOSProjectShaadi.xcodeproj/project.xcworkspace/contents.xcworkspacedata:1]: remove this file

File: iOSProjectShaadi.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist
Comment (Pulkit29) [inline iOSProjectShaadi.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:1]: remove this file

File: iOSProjectShaadi/Assets.xcassets/Onboarding/Ellipse 63.imageset/Contents.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (Pulkit29) [inline iOSProjectShaadi/Assets.xcassets/Onboarding/Ellipse 63.imageset/Contents.json]: make all assets single scale

File: iOSProjectShaadi/Assets.xcassets/Onboarding/closeButton.imageset/Vector.png
Comment (Pulkit29) [inline iOSProjectShaadi/Assets.xcassets/Onboarding/closeButton.imageset/Vector.png:1]: remove 1x

File: iOSProjectShaadi/Onboarding/Login/LoginViewController.swift [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .systemBackground
+                
+        otpButton.addTarget(self, action: #selector(otpButtonTapped), for: .touchUpInside)
+        passwordButton.addTarget(self, action: #selector(passwordButtonTapped), for: .touchUpInside)
+    }
+    
+    override func viewDidLayoutSubviews() {
+        setupConstraints()
```
Comment (Pulkit29) [inline iOSProjectShaadi/Onboarding/Login/LoginViewController.swift:82]: call super. viewDidLayoutSubviews first

File: iOSProjectShaadi/Onboarding/Login/LoginViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,10 @@
+//
+//  LoginViewModel.swift
+//  iOSProjectShaadi
+//
+//  Created by Aditya Vyavahare on 20/07/24.
+//
+
+class LoginViewModel {
```
Comment (Pulkit29) [inline iOSProjectShaadi/Onboarding/Login/LoginViewModel.swift:8]: use struct wherever possible

File: iOSProjectShaadi/Onboarding/Login/OTPViewController.swift [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .systemBackground
+            
+        otpStackView.delegate = self
+        otpStackView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(otpStackView)
+    }
+    
+    override func viewDidLayoutSubviews() {
```
Comment (Pulkit29) [inline iOSProjectShaadi/Onboarding/Login/OTPViewController.swift:84]: same

File: iOSProjectShaadi/Onboarding/Welcome/WelcomeView.swift [hunk 5]
```diff
@@ -0,0 +1,302 @@
+//
+//  WelcomeView.swift
+//  iOSProjectShaadi
+//
+//  Created by Rathi, Radhika on 24/07/2024.
+//
+
+import UIKit
+
+class WelcomeView: UIView {
```
Comment (Pulkit29) [inline iOSProjectShaadi/Onboarding/Welcome/WelcomeView.swift]: final

File: iOSProjectShaadi/Onboarding/Welcome/WelcomeView.swift [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+
+    // Get Started Button
+    getStartedButton.setTitle("Get Started", for: .normal)
+    getStartedButton.setTitleColor(.white, for: .normal)
+    getStartedButton.backgroundColor = UIColor(red: 239/255, green: 75/255, blue: 75/255, alpha: 1)
+    getStartedButton.layer.cornerRadius = 20
+    if let arrowImage = UIImage(named: "arrow.right")?.withRenderingMode(.alwaysOriginal) {
+      getStartedButton.setImage(arrowImage, for: .normal)
+    } else {
+      print("Arrow icon not found")
```
Comment (Pulkit29) [inline iOSProjectShaadi/Onboarding/Welcome/WelcomeView.swift]: use  debugPrint everywhere

File: iOSProjectShaadi/Onboarding/Welcome/WelcomeView.swift [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+      appleSignInButton.widthAnchor.constraint(equalToConstant: 57),
+      appleSignInButton.heightAnchor.constraint(equalToConstant: 60),
+
+      accountLabel.topAnchor.constraint(equalTo: getStartedButton.bottomAnchor, constant: 20),
+      accountLabel.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 20),
+      accountLabel.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -20),
+
+      loginButton.topAnchor.constraint(equalTo: getStartedButton.bottomAnchor, constant: 20),
+      loginButton.leadingAnchor.constraint(equalTo: accountLabel.trailingAnchor, constant: -80),
+      loginButton.bottomAnchor.constraint(equalTo: safeAreaLayoutGuide.bottomAnchor, constant: -20)
```
Comment (Pulkit29) [inline iOSProjectShaadi/Onboarding/Welcome/WelcomeView.swift]: break this method

File: iOSProjectShaadi/Onboarding/Welcome/WelcomeViewController.swift [hunk 8]
```diff
@@ -0,0 +1,73 @@
+//
+//  WelcomeViewController.swift
+//  iOSProjectShaadi
+//
+//  Created by Rathi, Radhika on 24/07/2024.
+//
+
+import UIKit
+
+class WelcomeViewController: UIViewController {
```
Comment (Pulkit29) [inline iOSProjectShaadi/Onboarding/Welcome/WelcomeViewController.swift]: final

File: iOSProjectShaadi/Onboarding/Welcome/WelcomeViewModel.swift [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+//  WelcomeViewModel.swift
+//  iOSProjectShaadi
+//
+//  Created by Rathi, Radhika on 31/07/2024.
+//
+
+import Foundation
+import GoogleSignIn
+
+class WelcomeViewModel {
```
Comment (Pulkit29) [inline iOSProjectShaadi/Onboarding/Welcome/WelcomeViewModel.swift]: try to use struct

Comment (Pulkit29) [inline iOSProjectShaadi/Onboarding/Welcome/WelcomeViewModel.swift]: create a protocol for public methods

File: iOSProjectShaadi/Utilities/OTPTextField.swift [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+//
+
+import UIKit
+
+protocol OTPDelegate: AnyObject {
+    //always triggers when the OTP field is valid
+    func didChangeValidity(isValid: Bool)
+}
+
+class OTPTextField: UITextField {
```
Comment (Pulkit29) [inline iOSProjectShaadi/Utilities/OTPTextField.swift]: final

File: iOSProjectShaadi/Utilities/OTPTextField.swift [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+    weak var previousTextField: OTPTextField?
+    weak var nextTextField: OTPTextField?
+    override public func deleteBackward() {
+        text = ""
+        previousTextField?.becomeFirstResponder()
+    }
+}
+
+class OTPStackView: UIStackView {
```
Comment (Pulkit29) [inline iOSProjectShaadi/Utilities/OTPTextField.swift]: final

File: iOSProjectShaadi/Assets.xcassets/Onboarding/Ellipse 63.imageset/Contents.json (hunk 1 above)
Comment (vvaditya23) [reply iOSProjectShaadi/Assets.xcassets/Onboarding/Ellipse 63.imageset/Contents.json]: Removed all the signup screens and related assets (will be added on `onboardingFlow-signup` br)

File: iOSProjectShaadi/Onboarding/Login/LoginViewController.swift (hunk 2 above)
Comment (vvaditya23) [reply iOSProjectShaadi/Onboarding/Login/LoginViewController.swift:82]: Done

File: iOSProjectShaadi/Onboarding/Login/LoginViewModel.swift (hunk 3 above)
Comment (vvaditya23) [reply iOSProjectShaadi/Onboarding/Login/LoginViewModel.swift:8]: I'll implement this once we decide on BE

File: iOSProjectShaadi/Onboarding/Login/OTPViewController.swift (hunk 4 above)
Comment (vvaditya23) [reply iOSProjectShaadi/Onboarding/Login/OTPViewController.swift:84]: Done

File: iOSProjectShaadi/Utilities/OTPTextField.swift (hunk 10 above)
Comment (vvaditya23) [reply iOSProjectShaadi/Utilities/OTPTextField.swift]: Done

File: iOSProjectShaadi/Utilities/OTPTextField.swift (hunk 11 above)
Comment (vvaditya23) [reply iOSProjectShaadi/Utilities/OTPTextField.swift]: Done

File: iOSProjectShaadi.xcodeproj/project.xcworkspace/contents.xcworkspacedata
Comment (vvaditya23) [reply iOSProjectShaadi.xcodeproj/project.xcworkspace/contents.xcworkspacedata:1]: Done

File: iOSProjectShaadi.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist
Comment (vvaditya23) [reply iOSProjectShaadi.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:1]: Done

**Pull request merged**